net/tcp/tcp_conn.c

```diff
@@ -549,16 +549,25 @@ FAR struct tcp_conn_s *tcp_alloc_conn(void)
 
   if (dq_peek(&g_free_tcp_connections) == NULL)
     {
+#if CONFIG_NET_TCP_MAX_CONNS > 0
+      if (dq_count(&g_active_tcp_connections) + CONFIG_NET_TCP_ALLOC_CONNS
```

Comment:
```suggestion
      if (dq_count(&g_active_tcp_connections) + CONFIG_NET_TCP_PREALLOC_CONNS
```

Reply:
No, this is correct.

The calculation adds the existing connections with those that are about to be allocated. The sum of them must not exceed the limit.

I don't see the meaning  of  `CONFIG_NET_TCP_PREALLOC_CONNS` here.

Reply:
Yes, it's my mistake.